js/search_bar.ts

```diff
@@ -0,0 +1,400 @@
+import type { RenderProps } from "@anywidget/types";
+import { html, css, PropertyValueMap } from "lit";
+import { property, query, queryAll } from "lit/decorators.js";
+import { classMap } from "lit/directives/class-map.js";
+
+import { legacyStyles } from "./ipywidgets_styles";
+import { LitWidget } from "./lit_widget";
+import { materialStyles } from "./styles";
+import { loadFonts } from "./utils";
+
+import { TabMode } from "./tab_panel";
+import { unsafeHTML } from "lit/directives/unsafe-html.js";
+
+export interface SearchTab {
+    search: string,
+    results: string[],
+    selected: string,
+    additional_html: string,
+}
+
+
+export interface SearchBarModel {
+    expanded: boolean;
+    tab_index: number;
+    name_address_model: string,
```

Comment:
I'm assuming you didn't want to use dictionaries here because of the dictionary change detection rules?

Reply:
Yeah... unfortunately, I couldn't come up with a clever or coherent way to make sure that dictionaries between the Python and JS layer triggered changes. It's usually possible in purely LitElement by making a copy, but the copy doesn't trigger a new dictionary update on the Python side, AFAICT.